wire-compiler/src/main/java/com/squareup/wire/CommandLineOptions.java

```diff
@@ -0,0 +1,137 @@
+package com.squareup.wire;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Scanner;
+import java.util.Set;
+
+
+final class CommandLineOptions {
+  public static final String PROTO_PATH_FLAG = "--proto_path=";
+  public static final String JAVA_OUT_FLAG = "--java_out=";
+  public static final String FILES_FLAG = "--files=";
+  public static final String REGISTRY_CLASS_FLAG = "--registry_class=";
+  public static final String ROOTS_FLAG = "--roots=";
+  public static final String NO_OPTIONS_FLAG = "--no_options";
+  public static final String ENUM_OPTIONS_FLAG = "--enum_options=";
+  public static final String SERVICE_WRITER_FLAG = "--service_writer=";
+  public static final String SERVICE_WRITER_OPT_FLAG = "--service_writer_opt=";
+
+  final String protoPath;
+  final List<String> sourceFileNames;
+  final List<String> roots;
+  final String javaOut;
+  final String registryClass;
+  final boolean emitOptions;
+  final Set<String> enumOptions;
+  final String serviceWriter;
+  final List<String> serviceWriterOptions;
+
+  CommandLineOptions(String protoPath, List<String> sourceFileNames, List<String> roots,
+      String javaOut, String registryClass, boolean emitOptions, Set<String> enumOptions,
+      String serviceWriter, List<String> serviceWriterOptions) {
+    this.protoPath = protoPath;
+    this.sourceFileNames = sourceFileNames;
+    this.roots = roots;
+    this.javaOut = javaOut;
+    this.registryClass = registryClass;
+    this.emitOptions = emitOptions;
+    this.enumOptions = enumOptions;
+    this.serviceWriter = serviceWriter;
+    this.serviceWriterOptions = serviceWriterOptions;
+  }
+
+  /**
```

Comment:
I was afraid it would only get out of date if we left it in WireCompiler.  I also considered moving the javadoc comment into a usage() string, but I don't think many folks actually invoke the compiler from the command line - they probably use a build tool  to invoke it.